sklearn/linear_model/_quantile.py

```diff
@@ -127,7 +129,11 @@ def fit(self, X, y, sample_weight=None):
             Returns self.
         """
         X, y = self._validate_data(
-            X, y, accept_sparse=False, y_numeric=True, multi_output=False
+            X,
+            y,
+            accept_sparse=["csr", "csc", "coo"],
```

Comment:
Since we later make a conversion from `coo` to `csr` then, we can remove supporting `coo` and it will be converted from now on.

```suggestion
            accept_sparse=["csr", "csc"],
```